src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/ProjectWorkspace.cs

```diff
@@ -0,0 +1,37 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.MSBuild;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.MSIdentity.CodeReaderWriter
+{
+    public class CodeAnaylsisHelper
+    {
+        private string _projectFilePath;
+        public CodeAnaylsisHelper(string projectFilePath)
+        {
+            _projectFilePath = projectFilePath;
+        }
+
+        private CodeAnalysis.Project? _codeAnalysisProject;
+        public CodeAnalysis.Project CodeAnalysisProject
+        {
+            get
+            {
+                if (_codeAnalysisProject == null)
+                {
+                    _codeAnalysisProject =  LoadCodeAnalysisProject().Result;
+                }
+                return _codeAnalysisProject;
+            }
+        }
+
+        private async Task<CodeAnalysis.Project> LoadCodeAnalysisProject()
+        {
+            var workspace = MSBuildWorkspace.Create();
+            var project = await workspace.OpenProjectAsync(_projectFilePath);
+            var projectWithFiles = project.WithAllSourceFiles();
```

Comment:
>WithAllSourceFiles [](http://example.com/codeflow?start=43&length=18)

when you do this addition of files to project - do you need to persist workspace?

Reply:
nope. Only for the DocumentEditor. I populate all the files in the project folder into a MSBuildWorkspace. Don't particularly care for all the compilation assemblies and project references.